src/core/dbus-manager.c

```diff
@@ -1565,11 +1565,13 @@ static int reply_unit_file_changes_and_free(
         unsigned i;
         int r;
 
-        if (n_changes > 0) {
-                r = bus_foreach_bus(m, NULL, send_unit_files_changed, NULL);
-                if (r < 0)
-                        log_debug_errno(r, "Failed to send UnitFilesChanged signal: %m");
-        }
+        for (i = 0; i < n_changes; i++)
+                if (unit_file_change_is_modification(changes[i].type)) {
```

Comment:
just an idea: Maybe add a unit_file_change_has_modification() call that takes the whole array and returns a bool? i.e. hide the for loop in install.c, not here...